node-commons/src/main/java/org/apache/iotdb/commons/utils/PathUtils.java

```diff
@@ -48,118 +49,131 @@ public static String[] splitPathToDetachedNodes(String path) throws IllegalPathE
     }
   }
 
-  public static void isLegalPath(String path) throws IllegalPathException {
+  public static String[] isLegalPath(String path) throws IllegalPathException {
     try {
-      PathNodesGenerator.splitPathToNodes(path);
+      return PathNodesGenerator.splitPathToNodes(path);
     } catch (PathParseException e) {
       throw new IllegalPathException(path);
     }
   }
 
   /**
-   * check whether measurement is legal according to syntax convention measurement can only be a
-   * single node name
+   * check whether measurement is legal according to syntax convention. Measurement can only be a
+   * single node name. The returned list is updated, could be different from the original list.
    */
-  public static void isLegalSingleMeasurementLists(List<List<String>> measurementLists)
-      throws MetadataException {
+  public static List<List<String>> checkIsLegalSingleMeasurementListsAndUpdate(
+      List<List<String>> measurementLists) throws MetadataException {
     if (measurementLists == null) {
-      return;
+      return null;
     }
     // use set to skip checking duplicated measurements
     Set<String> measurementSet = new HashSet<>();
+    List<List<String>> res = new ArrayList<>();
     for (List<String> measurements : measurementLists) {
-      checkLegalSingleMeasurementsAndSkipDuplicate(measurements, measurementSet);
+      res.add(checkLegalSingleMeasurementsAndSkipDuplicate(measurements, measurementSet));
     }
+    return res;
   }
 
   /**
-   * check whether measurement is legal according to syntax convention measurement can only be a
+   * check whether measurement is legal according to syntax convention. Measurement can only be a
    * single node name, use set to skip checking duplicated measurements
    */
-  public static void checkLegalSingleMeasurementsAndSkipDuplicate(
+  public static List<String> checkLegalSingleMeasurementsAndSkipDuplicate(
```

Comment:
we may need to change set to map, key is original string, value is formatted string.